### PR TITLE
[BUGFIX] Ensure datetime.time can be serialized to JSON

### PR DESCRIPTION
**Pull request by wookasz**
Adds `datetime.time` handling to the serialization code in utils. `datetime.time` is now serializable, and serialization is performed by converting it to its ISO format string.

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, visit our [community resources](https://docs.greatexpectations.io/docs/core/introduction/community_resources#contribute-code-or-documentation).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/test_ge_utils.py [hunk 1]
```diff
@@ -557,3 +559,21 @@ def test_convert_ndarray_float_to_datetime_tuple(
...
+    assert ret == {"t": "01:30:45"}
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "data", [pytest.param({"t": datetime.time(hour=1, minute=30, second=45)}, id="datetime.time")]
+)
+def test_ensure_json_serializable(data: dict):
+    ensure_json_serializable(data)
+    # Passes if no exception raised
```
Comment (joshua-stauffer) [inline tests/test_ge_utils.py]: tests look good, but they should probably live in `tests.test_convert_to_json_serializable`, unless there's a compelling reason to break the pattern.

**Review by joshua-stauffer — CHANGES_REQUESTED** [review]
looks good, just would recommend writing these tests next to the ones we already ahve

File: tests/test_ge_utils.py (hunk 1 above)
Comment (wookasz) [reply tests/test_ge_utils.py]: Thanks, I'll move. I didn't notice that file assuming we were following the pattern of test file per module

Comment (joshua-stauffer) [reply tests/test_ge_utils.py]: yeah, that would be super reasonable - at some point we could refactor this to be more logical. thanks for addressing!

**Review by joshua-stauffer — APPROVED** [review]
thanks!

**Pull request merged**